src/communication/xcConnection.ts

```diff
@@ -34,7 +29,9 @@ export class DefaultConnection implements Connection {
             log.debug("getModel request failed");
             log.debug(err);
         };
-        session.init(openListener, errorListener);
+        let closeListener = (_: CloseEvent) => {
```

Comment:
In my opinion, the good approach for handling disconnections is dealing with them just like other errors, you don't handle them, you let users know and deal with them. 

Disconnections should be handled just like that:

```
1. call the error callback, tells user connection is gone
2. disposes all resources (because the session we gave to users cannot be used anymore)
```
For an example of this kind of API, we have the websockets Rx API:

https://github.com/ReactiveX/rxjs/blob/736d48d2eb348bf8851264358c493a23872d0c8e/spec/observables/dom/webSocket-spec.ts#L355

Otherwise, if we really want to deal with it at API level, a good example of how to do it is by looking at the Slack NodeJS Wrapper:

https://github.com/slackapi/node-slack-sdk/blob/master/test/clients/rtm/retry-policy.js

You define the police once for all when you connect to the websocket (RTM).